Compilation failing on Java 9 early access build b90

Core compilation is failing due to modified type inference rules in JDK 9 early access build b90.

```
core/src/main/java/org/elasticsearch/common/collect/Iterators.java:31: error: cannot infer type arguments for ConcatenatedIterator<>
        return new ConcatenatedIterator<>(iterators);
               ^
  reason: cannot infer type-variable(s) T#1
    (varargs mismatch; Iterator<? extends T#2>[] cannot be converted to Iterator<? extends T#1>)
  where T#1,T#2 are type-variables:
    T#1 extends Object declared in class ConcatenatedIterator
    T#2 extends Object declared in method <T#2>concat(Iterator<? extends T#2>...)
```

--- a/core/src/main/java/org/elasticsearch/common/collect/Iterators.java
+++ b/core/src/main/java/org/elasticsearch/common/collect/Iterators.java
@@ -28,7 +28,8 @@ public class Iterators {
             throw new NullPointerException("iterators");
         }
 
-        return new ConcatenatedIterator<>(iterators);
+        // explicit generic type argument needed for type inference
+        return new ConcatenatedIterator<T>(iterators);
     }
 
     static class ConcatenatedIterator<T> implements Iterator<T> {

